src/main/java/io/github/opencubicchunks/cubicchunks/core/server/VanillaNetworkHandler.java

```diff
@@ -310,6 +324,9 @@ private static int writeData(PacketBuffer buf, ICube[] cubes, boolean hasSkyligh
     }
 
     public boolean hasCubicChunks(EntityPlayerMP player) {
+        if (!hasFML(player))    {
+            return false;
+        }
```

Comment:
is this necessary? it's essentially duplicating the work that hasCubicChunks is doing. And why does that code even run during FML handshake?

Reply:
```
net.minecraft.util.ReportedException: Adding entity to track
	at net.minecraft.entity.EntityTracker.track(EntityTracker.java:262) [EntityTracker.class:?]
	at net.minecraft.entity.EntityTracker.track(EntityTracker.java:180) [EntityTracker.class:?]
	at net.minecraft.world.ServerWorldEventHandler.onEntityAdded(ServerWorldEventHandler.java:43) [ServerWorldEventHandler.class:?]
	at net.minecraft.world.World.onEntityAdded(World.java:1341) [World.class:?]
	at net.minecraft.world.WorldServer.onEntityAdded(WorldServer.java:1175) [WorldServer.class:?]
	at net.minecraft.world.World.spawnEntity(World.java:1332) [World.class:?]
	at net.minecraft.world.WorldServer.spawnEntity(WorldServer.java:1121) [WorldServer.class:?]
	at io.github.opencubicchunks.cubicchunks.core.world.CubeWorldEntitySpawner.spawnCreatureTypeInAllChunks(CubeWorldEntitySpawner.java:219) [CubeWorldEntitySpawner.class:?]
	at io.github.opencubicchunks.cubicchunks.core.world.CubeWorldEntitySpawner.findChunksForSpawning(CubeWorldEntitySpawner.java:89) [CubeWorldEntitySpawner.class:?]
	at net.minecraft.world.WorldEntitySpawner.handler$zzp000$onSpawnMobs(WorldEntitySpawner.java:554) [WorldEntitySpawner.class:?]
	at net.minecraft.world.WorldEntitySpawner.findChunksForSpawning(WorldEntitySpawner.java) [WorldEntitySpawner.class:?]
	at net.minecraft.world.WorldServer.tick(WorldServer.java:212) [WorldServer.class:?]
	at net.minecraft.server.MinecraftServer.updateTimeLightAndEntities(MinecraftServer.java:831) [MinecraftServer.class:?]
	at net.minecraft.server.dedicated.DedicatedServer.updateTimeLightAndEntities(DedicatedServer.java:415) [DedicatedServer.class:?]
	at net.minecraft.server.MinecraftServer.tick(MinecraftServer.java:743) [MinecraftServer.class:?]
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:592) [MinecraftServer.class:?]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_162]
Caused by: java.lang.NullPointerException
	at io.github.opencubicchunks.cubicchunks.core.server.VanillaNetworkHandler.hasCubicChunks(VanillaNetworkHandler.java:318) ~[VanillaNetworkHandler.class:?]
	at net.minecraft.network.NetHandlerPlayServer.localvar$zcn000$onSendPacket(NetHandlerPlayServer.java:2575) ~[NetHandlerPlayServer.class:?]
	at net.minecraft.network.NetHandlerPlayServer.sendPacket(NetHandlerPlayServer.java) ~[NetHandlerPlayServer.class:?]
	at net.minecraft.entity.EntityTrackerEntry.updatePlayerEntity(EntityTrackerEntry.java:402) ~[EntityTrackerEntry.class:?]
	at net.minecraft.entity.EntityTrackerEntry.updatePlayerEntities(EntityTrackerEntry.java:511) ~[EntityTrackerEntry.class:?]
	at net.minecraft.entity.EntityTracker.track(EntityTracker.java:236) ~[EntityTracker.class:?]
	... 16 more
```

Reply:
that said, it could definitely be simplified (i hoisted it out into a separate function because of the other changes i've now reverted)